python/ray/data/aggregate.py

```diff
@@ -4,7 +4,8 @@
 from ray.util.annotations import PublicAPI
 from ray.data.block import T, U, KeyType, AggType
 
-AggregateOnT = Union[None, Callable[[T], Any], str]
+AggregateOnTBase = Union[Callable[[T], Any], str]
```

Comment:
I cannot comment on the code that's not changed so I just leave the comment here: the AggregateOnT used in this file should still have the old definition right? It's dataset.py that's using the new definition of AggregateOnT and do the validation and normalization, so we should move the new definition there?

Reply:
Ah yes great catch, this change was from an old implementation in this PR when `AggregateFn` could have one or more `on`s, which I decided to go away from. I'll update this!